modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/AbstractIgniteJoin.java

```diff
@@ -292,4 +292,8 @@ protected Mappings.TargetMapping buildTransposeMapping(boolean left2Right) {
                 (left2Right ? right : left).getRowType().getFieldCount()
         );
     }
+
+    @Override public String getRelTypeName() {
```

Comment:
Why don't have a default implementation in `IgniteRel` interface?

Reply:
Because all Ignite RelNodes inherit calcite `AbstractRelNode` (abstract class), which has a different `getRelTypeName()` implementation, therefore the default implementation from `IgniteRel` will not be used.

Reply:
Does it make sense to introduce abstract class `IgniteSingleRel` with overriden `getRelTypeName()` method?

Reply:
>   // It is not recommended to override this method, but sub-classes can do it at their own risk.
>   @Override public String getRelTypeName() {

😃

Reply:
> However, the better way is to use string constants

I missed this note, personally I agree, I updated the PR to use string constants :thinking: .